substrate/frame/revive/src/exec.rs

```diff
@@ -1323,6 +1328,24 @@ where
 	pub(crate) fn override_export(&mut self, export: ExportedFunction) {
 		self.top_frame_mut().entry_point = export;
 	}
+
+	#[cfg(all(feature = "runtime-benchmarks", feature = "riscv"))]
+	pub(crate) fn set_block_number(&mut self, block_number: BlockNumberFor<T>) {
+		self.block_number = block_number;
+	}
+
+	fn block_hash(&self, block_number: U256) -> Option<H256> {
+		let Ok(block_number) = BlockNumberFor::<T>::try_from(block_number) else {
+			return None;
+		};
```

Comment:
It is valid to request the hash of any U256 block number. So you request a block hash for a given number and either it is within the range of the last 256 blocks or it is not. If it overflows we just know upfront that the requested block number isn't in with the valid range and return `None`.

Or what error should it return in overflow?

Reply:
I would suggest trapping the execution on invalid input. For example, we trap with `BalanceConversionFailed` in case of supplied U256 is bigger than the balance. I guess you can just add a new error.

Reply:
Hmm you are right. Overflowing is the same as supplying a block in the future. Yeah let's keep it this way.